.clang-tidy-for-githooks

```diff
@@ -0,0 +1,41 @@
+# Keep in sync with .clang-tidy.
```

Comment:
We should try to think of some ideas to avoid the duplication. Here's one:

The `.clang-tidy` file is just a yaml file. If `clang-tidy` ignores fields it doesn't understand, we could add one in the existing `.clang-tidy` file, like `IgnoredCheckForFlutterGitHook` (or some better name). Then when running in the context of the pre-push hook, our `clang-tidy` wrapper can create a new `.clang-tidy` file under `/tmp`, and pass that to the `clang-tidy` `--config-file` flag.